### DISCUSSION TITLE
Mixed workload with specified percentage of Puts and Gets on etcd

### DISCUSSION THREAD

**Discussion by siri9209** [etcd-io/etcd#17218] [Q&A] [answered]
### What would you like to be added?

I started using this tool in one of our project and was able to submit puts and gets and seee the throughput and latency details.But wondering if there is any option to get mixed workload .

### Why is this needed?

it is good to have this feature as most of the customer workloads will be mixed as well.

**Comment by jmhbnz** [comment]
Hey @siri9209 - Thanks for raising this idea.  Can you please clarify which tool you are referring to?

**Comment by siri9209** [comment]
sorry, i was referring to etcdctl which was used for etcd performance benchmarking.

**Comment by jmhbnz** [comment, marked as answer]
If the usecase is for performance testing please consider using `rw-heatmaps` https://github.com/etcd-io/etcd/tree/main/tools/rw-heatmaps.

> etcd/tools/rw-heatmaps is the mixed read/write performance evaluation tool for etcd clusters.

**Comment by siri9209** [comment]
thanks for the info.
Is there any docker image that can i deploy on my setup and use these tools?

**Comment by jmhbnz** [comment]
> thanks for the info. Is there any docker image that can i deploy on my setup and use these tools?

`rw-heatmaps` is not included in any container image that I know of. It's basically just a couple of files if you have a look at the link I mentioned earlier so it would be trivial to add them to your own custom image if that is how you need to run it.

I'll convert this issue to a Q&A discussion as the feature you were originally looking for is available via `rw-heatmaps`.

**Reply by siri9209** [reply]
thank you.
